spring-cloud-generator/showcase/showcase-spring-starter/src/test/java/com/google/showcase/v1beta1/spring/EchoAutoConfigurationTests.java

```diff
@@ -0,0 +1,369 @@
+                  client.getSettings().getTransportChannelProvider();
+              assertThat(transportChannelProvider).isSameAs(transportChannelProviderBean);
+              assertThat(transportChannelProvider).isNotSameAs(mockTransportChannelProvider);
+            });
+  }
+
+  @Test
+  void testShouldUseDefaultTransportChannelProvider() {
+    this.contextRunner.run(
+        ctx -> {
+          EchoClient client = ctx.getBean(EchoClient.class);
+          TransportChannelProvider transportChannelProvider =
+              client.getSettings().getTransportChannelProvider();
+          TransportChannelProvider defaultTransportChannelprovider =
+              EchoSettings.defaultTransportChannelProvider();
+          assertThat(transportChannelProvider)
+              .usingRecursiveComparison()
+              .isEqualTo(defaultTransportChannelprovider);
+        });
+  }
+
+  //  @Test
+  //  void testCustomTransportChannelProviderSetToRest() {
+  //    this.contextRunner
+  //        .withPropertyValues("com.google.showcase.v1beta1.echo.use-rest=true")
+  //        .run(
+  //            ctx -> {
+  //              EchoClient client = ctx.getBean(EchoClient.class);
+  //              TransportChannelProvider transportChannelProvider =
+  //                  client.getSettings().getTransportChannelProvider();
+  //              TransportChannelProvider defaultHttpJsonTransportChannelprovider =
+  //                  EchoSettings.defaultHttpJsonTransportProviderBuilder().build();
+  //              assertThat(transportChannelProvider)
+  //                  .usingRecursiveComparison()
+  //                  .isEqualTo(defaultHttpJsonTransportChannelprovider);
+  //            });
+  //  }
+
+  @Test
+  void testQuotaProjectIdFromProperties() {
+    this.contextRunner
+        .withPropertyValues(
+            "com.google.showcase.v1beta1.echo.quota-project-id=" + SERVICE_OVERRIDE_CLIENT_ID)
+        .run(
+            ctx -> {
+              EchoClient client = ctx.getBean(EchoClient.class);
+              String quotaProjectId = client.getSettings().getQuotaProjectId();
+              assertThat(quotaProjectId).isEqualTo(SERVICE_OVERRIDE_CLIENT_ID);
+            });
+  }
+
+  @Test
+  void testExecutorThreadCountFromProperties() {
+    Integer customExecutorThreadCount = 3;
+    this.contextRunner
+        .withPropertyValues(
+            "com.google.showcase.v1beta1.echo.executor-thread-count="
+                + customExecutorThreadCount.toString())
+        .run(
+            ctx -> {
+              EchoClient client = ctx.getBean(EchoClient.class);
+              InstantiatingExecutorProvider backgroundExecutorProvider =
+                  ((InstantiatingExecutorProvider)
+                      client.getSettings().getBackgroundExecutorProvider());
+              assertThat(backgroundExecutorProvider.toBuilder().getExecutorThreadCount())
+                  .isEqualTo(customExecutorThreadCount);
+            });
+  }
+
+  //  @Test
```

Comment:
Similar error log as [above](https://github.com/GoogleCloudPlatform/spring-cloud-gcp/pull/1957/files#r1232822624).

Reply:
Resolved in https://github.com/GoogleCloudPlatform/spring-cloud-gcp/pull/1957/commits/0bdd26d59dd427b6f6fc5c635fab8203c0b1bca0.